src/utils/Utils.ts

```diff
@@ -438,4 +438,16 @@ export class Utils {
   static resolveAfter<T = void>(ms: number, returns?: T): Promise<T> {
     return new Promise(resolve => setTimeout(() => (returns !== undefined ? resolve(returns) : resolve()), ms));
   }
+
+  static reorderValuesByKeys<T, K extends string>(values: T[], order: K[], getKey: (value: T) => K, appendRemainingValues = true) {
+    const valuesMap = values.reduce<{ [key: string]: T }>((map, value) => ({ ...map, [getKey(value)]: value }), {});
+    const orderedValues: T[] = [];
+    order.forEach(keyToAppend => {
+      if (valuesMap[keyToAppend]) {
+        orderedValues.push(valuesMap[keyToAppend]);
+        delete valuesMap[keyToAppend];
+      }
+    });
+    return appendRemainingValues ? [...orderedValues, ..._.without(values, ...orderedValues)] : orderedValues;
```

Comment:
Originally, instead of `_.without(values, ...orderedValues)` I used `_.values(valuesMap)`. However, as much as the order of remaining values stayed the same that way, I find it unsafe to assume an object will always respect the order of key-value pairs.